lapce-ui/src/editor.rs

```diff
@@ -1484,6 +1485,128 @@ impl LapceEditor {
         }
     }
 
+    fn paint_selection_find(
+        ctx: &mut PaintCtx,
+        data: &LapceEditorBufferData,
+        screen_lines: &ScreenLines,
+    ) {
+        if !data.editor.content.is_file() {
+            return;
+        }
+        if screen_lines.lines.is_empty() {
+            return;
+        }
+        if !data.config.editor.highlight_selection_occurrences {
+            return;
+        }
+        let region = match &data.editor.cursor.mode {
+            lapce_core::cursor::CursorMode::Normal(offset) => {
+                lapce_core::selection::SelRegion::caret(*offset)
+            }
+            lapce_core::cursor::CursorMode::Visual {
+                start,
+                end,
+                mode: _,
+            } => lapce_core::selection::SelRegion::new(
+                *start.min(end),
+                data.doc.buffer().next_grapheme_offset(
+                    *start.max(end),
+                    1,
+                    data.doc.buffer().len(),
+                ),
+                None,
+            ),
+            lapce_core::cursor::CursorMode::Insert(selection) => {
+                *selection.last_inserted().unwrap()
+            }
+        };
+        let selection = data
+            .doc
+            .buffer()
+            .slice_to_cow(region.min()..region.max())
+            .to_string();
+
+        if !selection.is_empty() {
+            let cursor_offset = data.editor.cursor.offset();
+            let start_line = *screen_lines.lines.first().unwrap();
+            let end_line = *screen_lines.lines.last().unwrap();
+            let start = data.doc.buffer().offset_of_line(start_line);
+            let end = data.doc.buffer().offset_of_line(end_line + 1);
+            data.doc.update_selection_find(
+                &selection,
+                0,
+                data.doc.buffer().last_line(),
```

Comment:
I think this might cause a lag in a very large file. Why not search only the lines within `screen_lines`?

Reply:
Here is the reference 'find' logic (I know that it is a full document search for counting).
I tried to search in "screen_lines", but I didn't find a way to repaint. Can you give me some suggestions?

Reply:
Sorry, I made a mistake. When "screen_lines" changes, because the selection does not change, the search logic is not executed. 
But at the same time, I want to consider the situation that "screen_lines" and "selection" are unchanged, and I should not execute the search logic.
I'm looking for a good way to solve this problem